### ISSUE TITLE
Cannot change "Twin USB Joystick" to work

### ISSUE THREAD

**Issue by richard42** [mupen64plus/mupen64plus-user-issues#488]
Originally reported on Google Code with ID 489

```
Describe your system:
 - Windows XP Home SP3 32-bit
 - Mupen64Plus 1.99.5 Windows
 - Default plugins used

Describe the problem:
I edit the configuration for the "Twin USB Joystick" (a DualShock-to-USB adapter) in
AutoInputCfg, then load any ROM. The controls are still the default controls.

Please provide any additional information below.
I looked at the Wiki guide on how to change this and it doesn't help at all considering
it's for a version that existed 2 years ago and the Input-SDL-Control section in the
mupen64config file just flat out exist anymore, so it's hard to add something to it.
 Furthermore, as there are no help documentation files included with Mupen64Plus, nor
is there a -help printout in the command line for Mupen64Plus, troubleshooting this
problem is extremely difficult.
```

Reported by `written_testament@yahoo.com` on 2012-04-02 17:34:15

**Comment by richard42** [comment]
```
I should add that completely removing both the InputAutoCfg and mupen64plus configuration
files from the Mupen64Plus directory does absolutely nothing; all of the default configurations
for the Twin USB Joystick still exist, and I can still use it.  This leads to me to
believe that these files actually do nothing at all, since no part of the program actually
seems to use them in order to have a setup for any given controller.  And, on a note
relating to the removal of the mupen64plus configuration file, doing so actually stops
the spamming for about 60 Core Warnings for an eeprom file in the console, which is
an unrelated bug and a new issue but still a point to be raised.
```

Reported by `written_testament@yahoo.com` on 2012-04-02 17:41:48

**Comment by richard42** [comment]
```
Okay, well, I guess thanks anyway, I found out how to solve this.  For anyone else that's
curious: go into your application data folder and find the Mupen64Plus directory. 
Inside there will be a config file.  Find the Input-SDL-Control1 section that says
"please don't change" and ignore and change it anyway because this is literally the
only files Mupen64Plus seems to actually read controls from.  You can also change the
EEPROM directory while you're at it (which also says "please don't change) so you won't
get spammed 80 billion times in the console about it.  Also, I made my config file
read only after this, just to make sure Mupen64Plus wouldn't try to change it back,
but you may not have to.  However, I don't even care to try after all the work it has
been trying to solve this ridiculous problem.
```

Reported by `written_testament@yahoo.com` on 2012-04-06 04:14:47

**Comment by richard42** [comment]
```
Sorry for your troubles.  A few comments:

You can use the option "--help" to get a listing of the available command-line options.

If your eeprom directory is writable (it should be), you should only get those warnings
the first time you run a ROM.  I also intend to change this warning to an info message.

The "please don't change" remark in the config file is only referring to the joystick
name.  If you change the name, then the auto-config code will erase the stored config
and overwrite it.  You can change other parameters.  I will improve this comment to
make it more clear.
```

Reported by `richard42g` on 2012-04-06 04:27:16

**Comment by richard42** [comment]
```
I tried to reproduce your problem with the EEPROM messages with my Windows XP 32-bit
laptop and was not able to, using the 1.99.5 build.  Can you post the exact error messages
that you found, and the ROM that you used to produce these error messages?
```

Reported by `richard42g` on 2012-04-08 22:56:15

**Comment by richard42** [comment]
```
Awww Yeah! Eventually i can use my joystick on 1.99.5 . I was trying to solve this problem
from almost a week. The configuration file on Linux is in $HOME/.config/mupen64plus.
```

Reported by `francescobrozzu` on 2012-05-27 10:27:05

**Comment by richard42** [comment]
Reported by `sven@narfation.org` on 2013-12-18 11:52:55
- **Status changed**: `Fixed`
